font/src/directwrite/mod.rs

```diff
@@ -107,73 +243,27 @@ impl crate::Rasterize for DirectWriteRasterizer {
 
                     idx += 1;
                 }
-            },
+            }
         }?;
 
-        let face = font.create_font_face();
-        self.fonts.push(face);
+        let key = FontKey::next();
+        self.keys.insert(desc.clone(), key);
+        self.fonts.insert(key, font.into());
 
-        Ok(FontKey { token: (self.fonts.len() - 1) as u32 })
+        Ok(key)
     }
 
     fn get_glyph(&mut self, glyph: GlyphKey) -> Result<RasterizedGlyph, Error> {
-        let font = self.fonts.get(glyph.font_key.token as usize).ok_or(Error::FontNotLoaded)?;
-
-        let offset = GlyphOffset { advanceOffset: 0.0, ascenderOffset: 0.0 };
-
-        let glyph_index = *font
-            .get_glyph_indices(&[glyph.c as u32])
-            .first()
-            .ok_or_else(|| Error::MissingGlyph(glyph.c))?;
-        if glyph_index == 0 {
-            // The DirectWrite documentation states that we should get 0 returned if the glyph
-            // does not exist in the font
-            return Err(Error::MissingGlyph(glyph.c));
+        let loaded_font = self.get_loaded_font(glyph.font_key)?;
+
+        match self.rasterize_glyph(&loaded_font.face, glyph.size, glyph.c) {
+            Ok(glyph) => Ok(glyph),
+            Err(err @ Error::MissingGlyph(_)) => {
+                let fallback_font = self.get_fallback_font(&loaded_font, glyph.c).ok_or(err)?;
+                self.rasterize_glyph(&fallback_font.create_font_face(), glyph.size, glyph.c)
+            }
+            Err(err) => Err(err),
```

Comment:
```suggestion
            Err(err @ Error::MissingGlyph(_)) => {
                let fallback_font = self.get_fallback_font(&loaded_font, glyph.c).ok_or(err)?;
                self.rasterize_glyph(&fallback_font.create_font_face(), glyph.size, glyph.c)
            },
            result => result,
```

I'm not sure if there isn't an even better way to do this, but I feel like this makes things at least a little nicer.